custom_components/zwave_mqtt/binary_sensor.py

```diff
@@ -21,26 +22,261 @@
-    4: DEVICE_CLASS_HEAT,
-    5: DEVICE_CLASS_MOISTURE,
-    6: DEVICE_CLASS_SAFETY,
-    7: DEVICE_CLASS_SAFETY,
-    8: DEVICE_CLASS_POWER,
-    9: DEVICE_CLASS_PROBLEM,
-    10: DEVICE_CLASS_PROBLEM,
-    14: DEVICE_CLASS_SOUND,
-    15: DEVICE_CLASS_MOISTURE,
-    18: DEVICE_CLASS_GAS,
-}
+# Value Indexes for Notification CommandClass
+# TODO: Move to enum in the library
+NOTIFICATION_SMOKE_ALARM = 1
+NOTIFICATION_CARBON_MONOOXIDE = 2
+NOTIFICATION_CARBON_DIOXIDE = 3
+NOTIFICATION_HEAT = 4
+NOTIFICATION_WATER = 5
+NOTIFICATION_ACCESS_CONTROL = 6
+NOTIFICATION_HOME_SECURITY = 7
+NOTIFICATION_POWER_MANAGEMENT = 8
+NOTIFICATION_SYSTEM = 9
+NOTIFICATION_EMERGENCY = 10
+NOTIFICATION_CLOCK = 11
+NOTIFICATION_APPLIANCE = 12
+NOTIFICATION_HOME_HEALTH = 13
+NOTIFICATION_SIREN = 14
+NOTIFICATION_WATER_VALVE = 15
+NOTIFICATION_WEATHER = 16
+NOTIFICATION_IRRIGATION = 17
+NOTIFICATION_GAS = 18
+
+NOTIFICATION_TYPE = "index"
+NOTIFICATION_VALUES = "values"
+NOTIFICATION_DEVICE_CLASS = "device_class"
+NOTIFICATION_SENSOR_ENABLED = "enabled"
+
+
+# Translation from values in Notification CC to binary sensors
+# https://github.com/OpenZWave/open-zwave/blob/master/config/NotificationCCTypes.xml
+NOTIFICATION_SENSORS = [
+    {
+        # Index 1: Smoke Alarm - Value Id's 1 and 2
+        NOTIFICATION_TYPE: NOTIFICATION_SMOKE_ALARM,
+        NOTIFICATION_VALUES: [1, 2],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_SMOKE,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 1: Smoke Alarm - All other Value Id's
+        NOTIFICATION_TYPE: NOTIFICATION_SMOKE_ALARM,
+        NOTIFICATION_VALUES: [3, 4, 5, 6, 7, 8],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_SMOKE,
+        NOTIFICATION_SENSOR_ENABLED: False,
+    },
+    {
+        # Index 2: Carbon Monoxide - Value Id's 1 and 2
+        NOTIFICATION_TYPE: NOTIFICATION_CARBON_MONOOXIDE,
+        NOTIFICATION_VALUES: [1, 2],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_GAS,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 2: Carbon Monoxide - All other Value Id's
+        NOTIFICATION_TYPE: NOTIFICATION_CARBON_MONOOXIDE,
+        NOTIFICATION_VALUES: [4, 5, 7],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_GAS,
+        NOTIFICATION_SENSOR_ENABLED: False,
+    },
+    {
+        # Index 3: Carbon Dioxide - Value Id's 1 and 2
+        NOTIFICATION_TYPE: NOTIFICATION_CARBON_DIOXIDE,
+        NOTIFICATION_VALUES: [1, 2],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_GAS,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 3: Carbon Dioxide - All other Value Id's
+        NOTIFICATION_TYPE: NOTIFICATION_CARBON_DIOXIDE,
+        NOTIFICATION_VALUES: [4, 5, 7],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_GAS,
+        NOTIFICATION_SENSOR_ENABLED: False,
+    },
+    {
+        # Index 4: Heat - Value Id's 1, 2, 5, 6 (heat/underheat)
+        NOTIFICATION_TYPE: NOTIFICATION_HEAT,
+        NOTIFICATION_VALUES: [1, 2, 5, 6],
```

Comment:
Same question, are these four separate binary sensors? If so, I'm not so sure about that, especially when you consider the clear logic that has been added.

![image](https://user-images.githubusercontent.com/791794/80450491-7b3a0400-88d6-11ea-96f9-ed4e9dec1a48.png)

These appear to be defined as a single state variable. You could argue whether this is a binary_sensor or not, but if it is, I think it would be only two sensors. A single "overheat_detected" and another "underheat_detected". How can something be both overheated and underheated at the same time?

Reply:
These will be separate sensors. A sensor will be created for each value in the list if present.